src/js/jsx/sections/style/VectorAppearance.jsx

```diff
@@ -44,19 +43,16 @@ define(function (require, exports, module) {
         },
 
         render: function () {
-            var onlyTextLayers = this.props.document.layers.selected.every(function (layer) {
-                return layer.kind === layer.layerKinds.TEXT;
-            });
-
-            if (onlyTextLayers) {
+            if (this.props.uniformLayerKind === this.props.document.layers.byIndex(0).layerKinds.TEXT) {
```

Comment:
This seems like a reasonable idea, however, I think you should file that as a separate code cleanup task (as, the pattern I used here is the typical pattern for JSX throughout the rest of the code)